tensordict/tensordict.py

```diff
@@ -568,6 +577,22 @@ def pop(
                 )
         return out
 
+    def _check_meta(self, key: NESTED_KEY) -> MetaTensor:
+        _nested_key_type_check(key)
+        try:
+            if isinstance(key, tuple):
+                if len(key) > 1:
+                    return self.get(key[0])._check_meta(key[1:])
+                key = key[0]
+            return key in self._dict_meta
+        except KeyError:
+            # TODO: this error message will make more sense if `.keys()` returns nested
+            # keys rather than just the top level ones
+            raise KeyError(
+                f"key {key} not found in {self.__class__.__name__} with keys"
+                f" {sorted(self.keys(), key=str)}"
+            )
```

Comment:
The main problem here is when we call `_get_meta` we won't get `KeyError` because we will get into `__missing__` func and then `_make_meta` will be called. `_make_meta`  will try to create a new `MetaTensor`, but during initialisation `MetaTensor` will call `_requires_grad` and then it will iterate through all of nested objects and try to get their meta data (it will call `_get_meta` again) and here is the loop. I tried to find a simple way to stop this loop, and check option without [] operator fitted very well

Reply:
I see. Well what if we added something like the following to `_get_meta`.

```python
if key not in self._dict_meta:
    raise KeyError
```

That should prevent us from falling back to `__missing__` as you say, but also means we don't have to iterate over the nested structure twice?

Reply:
Sorry, I'm not sure I understood you correctly. In current code it's right behaviour to create new object of MetaTensor if we do not have it in self._dict_meta.  If we add 
```Python
if key not in self._dict_meta:
    raise KeyError
```
we will broke some new instances creation. Right now we check it only during `_requires_grad` process and assume that if we do not have a `key` it doesn't necessary to check grad for this item.